src/core/dxf/qgsdxfexport.cpp

```diff
@@ -190,7 +196,11 @@ QgsDxfExport::ExportResult QgsDxfExport::writeToFile( QIODevice *d, const QStrin
   }
 
   mTextStream.setDevice( d );
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
   mTextStream.setCodec( encoding.toLocal8Bit() );
+#else
+  mTextStream.setEncoding( QStringConverter::encodingForName( encoding.toLocal8Bit() ).value_or( QStringConverter::Utf8 ) );
```

Comment:
Cool, I didn't realise qt was using optional...

Reply:
Not an hour between your and my discovery of this